nestkernel/node_collection.h

```diff
@@ -490,7 +497,7 @@ NodeCollectionPTR operator+( NodeCollectionPTR lhs, NodeCollectionPTR rhs );
  * The composite type contains a collection of primitives which are not
  * contiguous and homogeneous with each other. If the composite is sliced, it
  * also holds information about what index to start at and which to end at, and
- * the step.
+ * the step. The endpoint is one past the last valid node.
```

Comment:
If find the documentation a bit confusing. Here you write that the endpoint is one past the last valid, in all comments below it says "... to end at", which sounds more like the "end" is included. Can we formulate this clearer?